### PR TITLE
Low speed

### PR DESCRIPTION
**Pull request by markmellors**
low speed dithering function added and tested ok

### CODE REVIEW

File: remote_control.py [hunk 1]
```diff
@@ -40,16 +42,21 @@ def run(self):
             try:
                 with ControllerResource() as joystick:
                     logging.info('Found a joystick and connected')
+                    left_counter=0
```
Comment (robberwick) [inline remote_control.py]: single space around assignment operator

File: remote_control.py [hunk 2]
```diff
@@ -110,3 +117,13 @@ def constrain(self, val, min_val, max_val):
...
         logging.info("RC challenge stopping")
         self.killed = True
+
+    def dither(self, counter, speed):
+        if self.deadband < abs(speed) < self.slow_speed:
+            speed = int(speed - math.copysign(1, speed))
+            counter += 1
+            if counter < self.boost_cycles:
+                speed = speed + int(math.copysign(self.slow_speed, speed))
+            if counter > (self.boost_cycles + self.boost_dwell):
```
Comment (robberwick) [inline remote_control.py]: `elif` here?  if `counter < self.boost_cycles`, then there's no point evaluating this condition.

**Review by robberwick — CHANGES_REQUESTED** [review]
Only a couple of very minor comments.

File: remote_control.py [hunk 3]
```diff
@@ -110,3 +117,13 @@ def constrain(self, val, min_val, max_val):
     def stop(self):
         logging.info("RC challenge stopping")
         self.killed = True
+
+    def dither(self, counter, speed):
```
Comment (robberwick) [inline remote_control.py:121]: We'd probably want to use this method when driving motors elsewhere in other challenges. How would we achieve that?

**Review by robberwick — COMMENTED** [review]
Additional comment regarding reusability of `dither()`

File: remote_control.py (hunk 2 above)
Comment (markmellors) [reply remote_control.py]: good point

File: remote_control.py (hunk 3 above)
Comment (markmellors) [reply remote_control.py:121]: I was thinking that we'd split RC into a motor driving module (with mixing, dither motor output functions in it) and a joystick input module. with dither being within the motor driving part and transparent to anything interacting with it. module may not be the right word here. I guess dither could be put into its own file/library to be used independently but I don't see a need for that.

Comment (robberwick) [reply remote_control.py:121]: That sounds like a reasonable approach 👍 

(and module was exactly the word I was looking for ;-) )

**Review by robberwick — APPROVED** [review]

**Comment by robberwick** [comment]
Looks good :+1:

**Pull request merged**